core/src/main/java/org/apache/gravitino/storage/relational/service/TableMetaService.java

```diff
@@ -183,26 +209,28 @@ public boolean deleteTable(NameIdentifier identifier) {
 
     Long tableId = getTableIdBySchemaIdAndName(schemaId, tableName);
 
+    AtomicInteger deleteResult = new AtomicInteger(0);
     SessionUtils.doMultipleWithCommit(
         () ->
-            SessionUtils.doWithoutCommit(
-                TableMetaMapper.class, mapper -> mapper.softDeleteTableMetasByTableId(tableId)),
+            deleteResult.set(
```

Comment:
Like the update, maybe the result can pass to the following operations to throw an exception 
 in advance.